### PR TITLE
Removed and replaced the XML window with an ImGui one

### PR DESCRIPTION
**Pull request by grimmier378**
* Removed the XML in game window.
* Added ImGui Window (same as the settings panel)
* Reworked the /pluginman command to toggle ImGui Window open | closed.
* Tossed a toggle button on the settings panel to toggle the main window.

### CODE REVIEW

File: MQ2PluginManager.cpp [hunk 1]
```diff
@@ -4,30 +4,37 @@
 	around until things worked. There may be some residual code that is not even required by the
 	plugin in here, but if it is, I'm unaware of it. Chances are some of these things I just deleted
 	entirely and tried to run after a succesful compile only to crash and therefore was readded.
+
+	July  9, 2024 Grimmier  -- Added ImGui Window to the mqsettings Panel.
+	July 30, 2024 Grimmier  -- Removed the in game XML window and replaced with a IMGUI window.
+							-- Added a toggle button to the mqsettings panel window.
+							-- Switched the /pluginman command to toggle the IMGUI window instead of the old XML.
```
Comment (brainiac) [inline MQ2PluginManager.cpp]: This is unnecessary

File: MQ2PluginManager.cpp [hunk 2]
```diff
@@ -4,30 +4,37 @@
...
 */
 
 #include <mq/Plugin.h>
 #include "CPluginTree.h"
-#include "CPluginToolWnd.h"
+#include "imgui/ImGuiUtils.h"
+#include <imgui_internal.h>
```
Comment (brainiac) [inline MQ2PluginManager.cpp]: Why is the internal header being included here?

File: MQ2PluginManager.cpp [hunk 3]
```diff
@@ -161,20 +129,29 @@ void DrawPluginManager_MQSettingsPanel()
...
 {
-	DestroyPluginWindow();
+	DrawGUI();
 }
 
-PLUGIN_API void OnReloadUI()
+PLUGIN_API void OnUpdateImGui()
 {
-	CreatePluginWindow();
+	if (GetGameState() == GAMESTATE_INGAME)
```
Comment (brainiac) [inline MQ2PluginManager.cpp]: I see no reason why this needs to be tied to game state anymore.

**Review by brainiac — COMMENTED** [review]
I left some comments/questions

File: MQ2PluginManager.cpp (hunk 2 above)
Comment (grimmier378) [reply MQ2PluginManager.cpp]: oops i was using easyfind plugin as a check to make sure i was creating the standalone window correctly.

File: MQ2PluginManager.cpp [hunk 4]
```diff
@@ -4,30 +4,37 @@
...
 static void DrawPluginManager_MQSettingsPanel();
 static void PluginManagerCommand(PlayerClient*, const char*);
+static bool s_showWindow = false;
+static void ImGui_ToggleWindow();
 
 //=-=-=-=-=- Plugin members =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=//
 PLUGIN_API void InitializePlugin()
 {
-	WriteChatColor("\aoLoading MQ2PluginManager v0.0.2 ...");
+	WriteChatColor("\aoLoading MQ2PluginManager v0.0.3 ...");
```
Comment (Sicprofundus) [inline MQ2PluginManager.cpp]: if we're going to leave the version on it here, we should put the version somewhere and then use it here

Comment (grimmier378) [reply MQ2PluginManager.cpp]: dropped it since it really doesn't do anything

File: MQ2PluginManager.cpp [hunk 5]
```diff
@@ -8,28 +8,30 @@
 
 #include <mq/Plugin.h>
 #include "CPluginTree.h"
-#include "CPluginToolWnd.h"
+#include "imgui/ImGuiUtils.h"
```
Comment (brainiac) [inline MQ2PluginManager.cpp]: What is using this header?

File: MQ2PluginManager.cpp [hunk 6]
```diff
@@ -161,20 +124,27 @@ void DrawPluginManager_MQSettingsPanel()
...
 
-PLUGIN_API void OnReloadUI()
+PLUGIN_API void OnUpdateImGui()
 {
-	CreatePluginWindow();
-}
+	if (!s_showWindow)
+		return;
 
-PLUGIN_API void SetGameState(int GameState)
```
Comment (brainiac) [inline MQ2PluginManager.cpp:174]: Changed my mind about this a little bit. Perhaps looking for additional feedback on this...

I don't know if we want to be immediately showing this at login by default, but would want it to appear in game by default like it did before. So, maybe make it toggleable at any point in the game, but only appear based base on last state when it was in game?

**Comment by grimmier378** [comment]
Yeah I was kind of thinking the same as I saw it appearing at login.

I can toss that check back in.

**Comment by brainiac** [comment]
> Yeah I was kind of thinking the same as I saw it appearing at login. I can toss that check back in.

Well I don't think you wrote all that I wrote. It would be nice if you could toggle the window from anywhere. The old gamestate check was because the in-game UI only works ingame. but ImGui works anywhere, so I don't think it should be as limited. Some thought needs to go into it.

File: MQ2PluginManager.cpp (hunk 5 above)
Comment (grimmier378) [reply MQ2PluginManager.cpp]: doh nothing it appears.

**Comment by grimmier378** [comment]
how do i get the slash command to work at char select?

```> /plugin pluginmanager
Loading MQ2PluginManager...
/pluginman Toggles the PluginManager UI Window
Plugin 'MQ2PluginManager' loaded.
> /pluginman
DoCommand - Couldn't parse '/pluginman'
```

I have some idea's but can't get the toggle to work outside of the game.

**Comment by brainiac** [comment]
> how do i get the slash command to work at char select?
> 
> ```
> Loading MQ2PluginManager...
> /pluginman Toggles the PluginManager UI Window
> Plugin 'MQ2PluginManager' loaded.
> > /pluginman
> DoCommand - Couldn't parse '/pluginman'
> ```
> 
> I have some idea's but can't get the toggle to work outside of the game.

Remove the optional params from AddCommand:
```
AddCommand("/Pluginman", PluginManagerCommand, false, false, true);
```

from the doc comments:
```c++
/**
 * Adds a new chat command. If the command conflicts with an existing EverQuest command, then
 * this command will take precedence. If the ocmmand conflicts with another MacroQuest command,
 * then this call will fail.
 *
 * @param command The name of the command to register, including a leading forward slash, e.g. "/hello"
 * @param handler The function that will be call when the command is executed.
 * @param eq If true, the handler is ignored and the command will be forwarded to EQ.
 * @param parse If true, any macro expressions will be parsed before passing the text to the handler.
 * @param inGame If true, the command will only be registered while in the game world.
 * @return True if the command is successfully added.
 */
bool AddCommand(std::string_view command, MQCommandHandler handler, bool eq = false, bool parse = true, bool inGame = false);
```

**Pull request merged**